modules/account-tokenizer/src/lib.rs

```diff
@@ -0,0 +1,385 @@
+
+		/// Pallet's account - used to mint and burn NFT.
+		#[pallet::constant]
+		type PalletAccount: Get<Self::AccountId>;
+
+		/// Treasury's account. Fees and penalties are transferred to the treasury.
+		#[pallet::constant]
+		type TreasuryAccount: Get<Self::AccountId>;
+
+		/// The XcmInterface to communicate with the relaychain via XCM.
+		type XcmInterface: ProxyXcm<Self::AccountId>;
+
+		/// Origin used by Oracles. Used to relay information from the Relaychain.
+		type OracleOrigin: EnsureOrigin<Self::Origin>;
+
+		type NFTInterface: Inspect<Self::AccountId, ClassId = Self::ClassId, InstanceId = Self::TokenId>
+			+ Mutate<Self::AccountId>
+			+ InspectExtended<Self::AccountId>;
+	}
+
+	#[pallet::error]
+	pub enum Error<T> {
+		/// The AccountToken's NFT class ID as not yet been set.
+		NFTClassIdNotSet,
+		/// The given account's NFT is already issued.
+		AccountTokenAlreadyMinted,
+		/// The given account's already been requested to mint.
+		AccountAlreadyRequestedMinted,
+		/// The mint request for the given account is not found.
+		MintRequestNotFound,
+		/// The confirmed owner and the mint requester isn't the same.
+		MintRequestDifferentFromOwner,
+		/// The account's NFT token cannot be found.
+		AccountTokenNotFound,
+		/// The caller is unauthorized to make this transaction.
+		CallerUnauthorized,
+	}
+
+	#[pallet::event]
+	#[pallet::generate_deposit(fn deposit_event)]
+	pub enum Event<T: Config> {
+		/// The class ID of Account Tokens has been set
+		NFTClassIdSet {
+			class_id: ClassIdOf<T>,
+		},
+		/// The mint fee has been set
+		MintFeeSet {
+			mint_fee: Balance,
+		},
+		/// The burn fee has been set
+		BurnFeeSet {
+			burn_fee: Balance,
+		},
+		/// The deposit amount for request_mint has been set
+		RequestMintDepositSet {
+			deposit: Balance,
+		},
+		/// The user has requested to mint a Account Token NFT.
+		MintRequested {
+			account: T::AccountId,
+			who: T::AccountId,
+		},
+		MintRequestRejected {
+			account: T::AccountId,
+			who: T::AccountId,
+		},
+		/// A NFT is minted to the owner of an account on the Relaychain.
+		AccountTokenMinted {
+			account: T::AccountId,
+			owner: T::AccountId,
+			token_id: T::TokenId,
+		},
+		/// The account token is burned, the control of the `account` on the relaychain is
+		/// relinquished to `new_owner`.
+		AccountTokenBurned {
+			account: T::AccountId,
+			owner: T::AccountId,
+			token_id: T::TokenId,
+			new_owner: T::AccountId,
+		},
+	}
+
+	/// Stores the NFT's class ID. Settable by authorized Oracle. Used to mint and burn PRTs.
+	#[pallet::storage]
+	#[pallet::getter(fn nft_class_id)]
+	type NFTClassId<T: Config> = StorageValue<_, ClassIdOf<T>, OptionQuery>;
+
+	/// Stores accounts that are already minted as an NFT.
+	/// Storage Map: Tokenized Account Id  => NFT Token ID
+	#[pallet::storage]
+	#[pallet::getter(fn minted_account)]
+	type MintedAccount<T: Config> = StorageMap<_, Twox64Concat, T::AccountId, T::TokenId, OptionQuery>;
```

Comment:
This will prevent "double mint". Otherwise a user can apply to mint 2 NFTs of the same account.
This also records the token ID, so we do not need to search for it when the user is burning it.